test/livebook_teams/hubs/team_client_test.exs

```diff
@@ -542,13 +565,431 @@ defmodule Livebook.Hubs.TeamClientTest do
+        )
+
+      {:ok, pid} = TeamClient.start_link(team)
+      assert_receive {:hub_connected, ^id}
+      refute_receive {:deployment_group_created, ^deployment_group}
+
+      send(pid, {:event, :user_connected, user_connected})
+      assert_receive {:deployment_group_created, ^deployment_group}
+
+      updated_deployment_group = %{deployment_group | mode: "online"}
+
+      updated_livebook_proto_deployment_group =
+        LivebookProto.DeploymentGroup.new!(
+          id: to_string(updated_deployment_group.id),
+          name: updated_deployment_group.name,
+          mode: to_string(updated_deployment_group.mode),
+          secrets: []
+        )
+
+      user_connected =
+        LivebookProto.UserConnected.new!(
+          name: team.hub_name,
+          secrets: [],
+          file_systems: [],
+          deployment_groups: [updated_livebook_proto_deployment_group]
+        )
+
+      send(pid, {:event, :user_connected, user_connected})
+      assert_receive {:deployment_group_updated, ^updated_deployment_group}
+
+      refute deployment_group in TeamClient.get_deployment_groups(team.id)
+      assert updated_deployment_group in TeamClient.get_deployment_groups(team.id)
+    end
+
+    test "deletes the deployment group", %{user: user, node: node} do
+      team = build_team_hub(user, node)
+      id = team.id
+
+      deployment_group =
+        build(:deployment_group,
+          id: "1",
+          name: "sleepy-cat-#{System.unique_integer([:positive])}",
+          mode: "offline",
+          hub_id: id,
+          secrets: []
+        )
+
+      livebook_proto_deployment_group =
+        LivebookProto.DeploymentGroup.new!(
+          id: to_string(deployment_group.id),
+          name: deployment_group.name,
+          mode: to_string(deployment_group.mode),
+          secrets: []
+        )
+
+      user_connected =
+        LivebookProto.UserConnected.new!(
+          name: team.hub_name,
+          secrets: [],
+          file_systems: [],
+          deployment_groups: [livebook_proto_deployment_group]
+        )
+
+      {:ok, pid} = TeamClient.start_link(team)
+      assert_receive {:hub_connected, ^id}
+      refute_receive {:deployment_group_created, ^deployment_group}
+
+      send(pid, {:event, :user_connected, user_connected})
+      assert_receive {:deployment_group_created, ^deployment_group}
+
+      user_connected =
+        LivebookProto.UserConnected.new!(
+          name: team.hub_name,
+          secrets: [],
+          file_systems: [],
+          deployment_groups: []
+        )
+
+      send(pid, {:event, :user_connected, user_connected})
+      assert_receive {:deployment_group_deleted, ^deployment_group}
+
+      refute deployment_group in TeamClient.get_deployment_groups(team.id)
+    end
+  end
+
+  describe "agent connected event" do
+    setup do
+      agent_name = "chonky-cat-#{System.unique_integer([:positive])}"
+      Application.put_env(:livebook, :agent_name, agent_name)
```

Comment:
We shouldn't set global config in async tests as it may lead to race conditions. Ideally we want the tests to be async though, so perhaps specify agent name per teams client. See my comment in `application.ex` though, if we moved agent name to the struct, we wouldn't need to rely on global state.

Reply:
As we are going to use a separate environment variable for the name of the Hub, perhaps define an agent name in `test_helper` and use it in all tests that make sense (it will only be used when it is an agent test)

Reply:
Ah, I assumed the name needs to be unique, yeah `test_helper` totally works then!